solidity/contracts/libraries/operator/Groups.sol

```diff
@@ -23,8 +26,10 @@ library Groups {
         // The value is set when the operator contract is added.
         uint256 relayEntryTimeout;
 
+        // Mapping of `groupPubKey` to flagged `groupIndex`
+        mapping (bytes => uint256) groupIndices;
         Group[] groups;
-        uint256[] terminatedGroups;
+        uint256[] activeTerminatedGroups;
```

Comment:
We keep the indices of non-expired terminated groups there. When they expire, they're removed from the array, which meant that we were removing all records of those groups' termination. The array being called `terminatedGroups` confused me because it doesn't contain all terminated groups, but only the ones that haven't expired yet.

Reply:
Yeah, you are right. Though it depends on the perspective. I was treating `terminated` as a separate state that's why it was probably less confusing to me. Let's leave it as you proposed.